src/tests/conformance/handshake.rs

```diff
@@ -32,3 +36,38 @@ async fn c001_handshake_when_node_receives_connection() {
     synthetic_node.shut_down().await;
     node.stop().unwrap();
 }
+
+#[tokio::test]
+async fn c002_handshake_when_node_initiates_connection() {
+    // ZG-CONFORMANCE-002
+
+    // Create a synthetic node and enable handshaking.
+    let synthetic_node = SyntheticNodeBuilder::default()
+        .build()
+        .await
+        .expect("Unable to build a synthetic node");
+
+    // Spin up a node instance.
+    let target = TempDir::new().expect("Couldn't create a temporary directory");
+    let mut node = Node::builder()
+        .initial_peers(vec![synthetic_node.listening_addr().unwrap()])
```

Comment:
```suggestion
        .initial_peers(&[synthetic_node.listening_addr().unwrap()])
```
(with that one aforementioned API change)